yogstation/code/modules/donor/unique_donator_items.dm

```diff
@@ -0,0 +1,542 @@
+
+To create a new unique donator item, create a new datum/donator_item
+
+@version 1.0
+@author Kmc2000
+
+*/
+
+GLOBAL_LIST_INIT(donor_pdas, list("Normal", "Transparent", "Pip Boy", "Rainbow"))
+
+GLOBAL_DATUM_INIT(donator_gear, /datum/donator_gear_resources, new)
+
+/client/proc/custom_donator_item()
+	if(!is_donator(src))
+		to_chat(src, "<span class='warning'>You're not a donator! To access this feature, considering donating today!</span>")
+		return
+	GLOB.donator_gear.ui_interact(usr)//datum has a tgui component, here we open the window
+
+/datum/donator_gear_resources
+	var/name = "Unique Donator Items Controller"
+	var/list/donor_items = list()
+
+//We allow any state here because a dead person can set their donator hat and it won't really change a whole lot.
+/datum/donator_gear_resources/ui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, \
+force_open = FALSE, datum/tgui/master_ui = null, datum/ui_state/state = GLOB.always_state)//ui_interact is called when the client verb is called.
+
+	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
+	if(!ui)
+		ui = new(user, src, ui_key, "DonorGear", "Donator Gear Setup", 300, 300, master_ui, state)
+		ui.open()
+
+/datum/donator_gear_resources/ui_act(action, list/params, datum/tgui/ui, datum/ui_state/state)
+	. = ..()
+	usr = (istype(usr, /client)) ? usr : usr.client
+	var/datum/donator_gear/DG = locate(params["target"])
+	if(!DG)
+		return
+	DG.equip(usr)
+
+/datum/donator_gear_resources/ui_data(mob/user)
+	var/list/data = list()
+	var/list/items_info = list()
+	items_info["hats"] = list()
+	items_info["items"] = list()
+	data["items_info"] = items_info
+	for(var/datum/donator_gear/S in GLOB.donator_gear.donor_items)
+		if(S && lowertext(S.ckey) == lowertext(user?.client?.ckey) || !S.ckey) //Nulled out Ckey entries are assumed to be owned by all donators.
+			var/list/item_info = list()
+			item_info["name"] = S.name
+			item_info["selected"] = "[user?.client?.prefs.donor_item]" == "[S.unlock_path]" || "[user?.client?.prefs.donor_hat]" == "[S.unlock_path]"
+			item_info["id"] = "\ref[S]"
+			//Bit of sorting on the UI to make it nicer to look at.
+			if(S.slot == SLOT_HEAD)
+				var/list/L = items_info["hats"] //To dodge the annoying linting error. This one really irks me.
+				items_info["hats"][++L.len] = item_info
+			else
+				var/list/L = items_info["items"] //To dodge the annoying linting error. This one really irks me.
+				items_info["items"][++L.len] = item_info
+	data["items_info"] = items_info
+	return data
+
+//Constructor for the donator item controller. Ensures that the donor items list is populated.
+
+/datum/donator_gear_resources/New()
+	. = ..()
+	for(var/Stype in subtypesof(/datum/donator_gear))
+		var/datum/donator_gear/S = new Stype
+		if(!S.unlock_path)
+			message_admins("WARNING: [S] has no unlock path, clearing it out.")
+			qdel(S)
+			continue 
+		donor_items += S
+
+/datum/donator_gear
```

Comment:
Move these // comments to above the thing they are meant to be commenting on, and add another / so it becomes

///penis wenis
var/bagel